xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-filters/xwiki-platform-notifications-filters-api/src/main/java/org/xwiki/notifications/filters/internal/DefaultNotificationFilterManager.java

```diff
@@ -207,6 +214,29 @@ public Stream<NotificationFilter> getToggleableFilters(Collection<NotificationFi
         return filters.stream().filter(filter -> filter instanceof ToggleableNotificationFilter);
     }
 
+    @Override
+    public List<ToggleableNotificationFilter> getToggleableFilters(UserReference userReference)
+        throws NotificationException
+    {
+        DocumentReference userDoc = this.documentReferenceUserReferenceSerializer.serialize(userReference);
+        List<ToggleableNotificationFilter> result = new ArrayList<>();
+        getAllFilters(userDoc, false)
+            .stream().filter(filter -> filter instanceof ToggleableNotificationFilter)
+            .forEach(item -> result.add((ToggleableNotificationFilter) item));
+        return result;
+    }
+
+    @Override
+    public List<ToggleableNotificationFilter> getToggleableFilters(WikiReference wikiReference)
+        throws NotificationException
+    {
+        List<ToggleableNotificationFilter> result = new ArrayList<>();
+        getAllFilters(wikiReference)
+            .stream().filter(filter -> filter instanceof ToggleableNotificationFilter)
+            .forEach(item -> result.add((ToggleableNotificationFilter) item));
+        return result;
```

Comment:
Slightly more idiomatic (can be applied on the next method as well).
```suggestion
        DocumentReference userDoc = this.documentReferenceUserReferenceSerializer.serialize(userReference);
        return getAllFilters(userDoc, false)
            .stream()
            .filter(filter -> filter instanceof ToggleableNotificationFilter)
            .map(filter -> (ToggleableNotificationFilter) filter)
            .toList();
```

Reply:
Thanks I missed your comment but I actually applied same solution in new version :)